### ISSUE TITLE
docs: cutting a release

### ISSUE THREAD

**Issue by robjloranger** [gomods/athens#567]
continuing the discussion from: https://github.com/gomods/athens/pull/547#discussion_r212680314

We need to outline the steps required for a maintainer to cut a new release of either the proxy or Olympus.

**Comment by robjloranger** [comment]
Here are some thoughts from slack, after we added auto generation of binaries with GoReleaser.

...
I also saw it supports using a custom release notes markdown file. I wonder if we should do releases as a PR, this would just be a release notes file and new tag. That way we can control the notes contents. As it is if we allow the automatic generation it will just build a log from commit messages.
Or we can enable `draft` for releases and they will be added as such, not immediately released. This would allow editing of the release after generation.
...

I am leaning towards the draft option, this will auto populate the release and save it as a draft on GH. Allowing maintainers to work on the notes as needed before publishing the release.

**Comment by robjloranger** [comment]
I have a PR open to use the draft option, #621

**Comment by robjloranger** [comment]
I think it should stay open, the doc would be a general outline of how to increment tags and when. What to include and how to format the changelog. As well with the draft feature, a maintainer would push a tag first to generate the binaries and draft. Then edit the release doc. Maybe releases require two maintainers? One to approve.

**Comment by robjloranger** [comment]
I could incorporate it into #570

**Comment by DrPsychick** [comment]
Yes, the docs are outdated: https://github.com/gomods/athens/blob/main/DEVELOPMENT.md#for-people-doing-a-release

Current procedure is very simple:
- agree in Slack with maintainers what goes into the release & merge those PRs -> changes determines the semver (major/minor/patch bump)
- create a release in GitHub directly, generate the Changelog and clean it up if needed -> publish and Actions will do the rest.
